src/components/EditActivityForm.js

```diff
@@ -136,15 +136,24 @@ class EditActivityForm extends Component {
       activityToEdit.patch(activity);
       swal('Success', 'Activity saved', 'success');
     } else {
-      this.submitTask = store.createStudentActivity(student, schoolYear, fields);
-      const activity = await this.submitTask;
-      if (onCreateActivity) onCreateActivity(activity);
-      swal('Success', 'Activity created', 'success');
+      if (students) {
+        students.forEach((async (student) => this.createStudentActivity(student, schoolYear, fields)))
```

Comment:
I think `await Promise.all(students.map(student => this.createStudentActivity(..fitb..)))` would be a better approach. Alternatively, creating a new endpoint that takes in one or more students and process them on the server would be next level awesome, but this is fine as it is.

Reply:
Definitely! I can make a separate PR in the server repo to add that functionality in.